code/modules/admin/tickets/adminticketsUI.dm

```diff
@@ -0,0 +1,173 @@
+			else
+				continue
+	else  if(tab == ADMIN_TICKET_RESOLVED)
+		dat += "<h2>Resolved Tickets</h2>"
+		for(var/T in allTickets)
+			ticket = T
+			if(ticket.ticketState == ADMIN_TICKET_RESOLVED)
+				dat += "<tr style='[trStyle]'><td style ='[tdStyleleft]'><a href='?src=[UID()];resolve=[ticket.ticketNum]'>Resolve</a><a href='?src=[UID()];details=[ticket.ticketNum]'>Details</a> <br /> #[ticket.ticketNum] ([ticket.timeOpened]) </td><td style='[tdStyle]'><b>[ticket.title]</td></tr>"
+			else
+				continue
+	else if(tab == ADMIN_TICKET_CLOSED)
+		dat += "<h2>Closed Tickets</h2>"
+		for(var/T in allTickets)
+			ticket = T
+			if(ticket.ticketState == ADMIN_TICKET_CLOSED)
+				dat += "<tr style='[trStyle]'><td style ='[tdStyleleft]'><a href='?src=[UID()];resolve=[ticket.ticketNum]'>Resolve</a><a href='?src=[UID()];details=[ticket.ticketNum]'>Details</a> <br /> #[ticket.ticketNum] ([ticket.timeOpened]) </td><td style='[tdStyle]'><b>[ticket.title]</td></tr>"
+			else
+				continue
+
+	dat += "</table></body>"
+
+	return dat
+
+/datum/adminTicketHolder/proc/showUI(var/client/C, var/tab)
+	var/dat = null
+	dat = returnUI(tab)
+	var/datum/browser/popup = new(usr, "admintickets", "Admin Tickets", 1400, 600)
+	popup.set_content(dat)
+	popup.open()
+
+/datum/adminTicketHolder/proc/showDetailUI(var/client/C, var/ticketID)
+	var/datum/admin_ticket/T = globAdminTicketHolder.allTickets[ticketID]
+	var/status = "[T.state2text()]"
+
+	var/dat = "<h1>Admin Tickets</h1>"
+
+	dat +="<a href='?src=[UID()];refresh=1'>Show All</a><a href='?src=[UID()];refreshdetail=[T.ticketNum]'>Refresh</a>"
+
+	dat += "<h2>Ticket #[T.ticketNum]</h2>"
+
+	dat += "<h3>[T.clientName] / [T.mobControlled] opened this ticket at [T.timeOpened] at location [T.locationSent]</h3>"
+	dat += "<h4>Ticket Status: <font color='red'>[status]</font>"
+	dat += "<table style='width:950px; border: 3px solid;'>"
+	dat += "<tr><td>[T.title]</td></tr>"
+
+	if(T.content.len > 1)
+		for(var/i = 2, i <= T.content.len, i++)
+			dat += "<tr><td>[T.content[i]]</td></tr>"
+
+	dat += "</table><br /><br />"
+	dat += "<a href='?src=[UID()];detailreopen=[T.ticketNum]'>Re-Open</a><a href='?src=[UID()];detailresolve=[T.ticketNum]'>Resolve</a><br /><br />"
+
+	if(!T.adminAssigned)
+		dat += "No admin assigned to this ticket - <a href='?src=[UID()];assignadmin=[T.ticketNum]'>Take Ticket</a><br />"
+	else
+		dat += "[T.adminAssigned] is assigned to this Ticket. - <a href='?src=[UID()];assignadmin=[T.ticketNum]'>Take Ticket</a><br />"
+
+	if(T.lastAdminResponse)
+		dat += "<b>Last Admin Response:</b> [T.lastAdminResponse] at [T.lastResponseTime]"
+	else
+		dat +="<font color='red'>No Admin Response</font>"
+
+	dat += "<br /><br />"
+
+	dat += "<a href='?src=[UID()];detailclose=[T.ticketNum]'>Close Ticket</a>"
+
+	var/datum/browser/popup = new(usr, "adminticketsdetail", "Admin Ticket #[T.ticketNum]", 1000, 600)
```

Comment:
If possible, it would be better to have a strict `user` argument on these two UI procs.